extensions/2.0/Khronos/KHR_lights_shadows/Readme.md

```diff
@@ -0,0 +1,107 @@
+# KHR\_lights\_shadows
+
+## Contributors
+
+* Takahiro Aoyagi, Mozilla, [@takahirox](https://github.com/takahirox)
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec and
+[`KHR_lights_punctual`](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_lights_punctual).
+
+## Overview
+
+This `KHR_lights_shadows` extension defines what meshes cast or receive shadows
+and what lights cast shadows against
+[`KHR_lights_punctual`](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_lights_punctual).
+
+Rendering with shadows needs a lot of calculation. It is a common technique to
+select objects which cast or receive shadows in a scene.
+`KHR_lights_punctual` extension has the capability of defining lights but doesn't
+have the capability of defining shadows.
+
+This `KHR_lights_shadows` extension provides a way to let applications or
+engines know what meshes should cast or receive shadows and what lights should
+cast shadows.
+
+## Cast/Receive shadows for Meshes
+
+Whether a mesh casts shadows and whether it receives shadows are defined by
+adding `extensions.KHR_lights_shadows` property to a `mesh.primitive` definition

```

Comment:
How material settings are connected with shadow caster?

Some rare use cases:
1. It is correct that facingness of casting meshes is defined by [**doubleSided** of **material**](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_material_doublesided)? In theory, separating facingness of material and shadow casting can improve performance. But probably, it is a rare case (for example, 3d scans).
2. Is it correct that if we want to render a shadow without rendering caster then set `castShadows: true` and don't set [**material**](https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_mesh_primitive_material) of primitive?